### PR TITLE
Add token authentication to Terraform provider

### PR DESCRIPTION
**Pull request by adriansuarez**
This change adds token authentication to the Terraform provider and enables token authentication on test runs where the server is configured to generate tokens.

### CODE REVIEW

File: deploy/k8s/env.sh [hunk 1]
```diff
@@ -11,14 +11,24 @@ hostname="$(kubectl get service ingress-nginx-controller -o jsonpath='{.status.l
...
-NUODB_CP_USER=system/admin
-NUODB_CP_PASSWORD="$(kubectl get secret dbaas-user-system-admin -o jsonpath='{.data.password}' | base64 -d)"
+export NUODB_CP_URL_BASE="http://$hostname:$port/nuodb-cp"
+export NUODB_CP_USER=system/admin
+export NUODB_CP_PASSWORD="$(kubectl get secret dbaas-user-system-admin -o jsonpath='{.data.password}' | base64 -d)"
+
+# Use token authentication if server is configured to generate tokens
+if [ ! "$(kubectl get secrets nuodb-cp-runtime-config -o jsonpath='{.data.secretPassword}' --ignore-not-found)" = "" ]; then
+    NUODB_CP_TOKEN="$(nuodb-cp httpclient POST login --jsonpath token --unquote)"
+    if [ ! "$NUODB_CP_TOKEN" = "" ]; then
```
Comment (sivanov-nuodb) [inline deploy/k8s/env.sh:21]: Nit: Why not `[ "$NUODB_CP_TOKEN" != "" ]` or `[ -n "$NUODB_CP_TOKEN" ]` which reads better?

**Review by sivanov-nuodb — APPROVED** [review]
Looks good.

File: deploy/k8s/env.sh (hunk 1 above)
Comment (adriansuarez) [reply deploy/k8s/env.sh:21]: I don't remember why I got into the habit of doing it this way, but I'll change it to `[ -n "$NUODB_CP_TOKEN" ]`.

File: deploy/k8s/env.sh [hunk 2]
```diff
@@ -11,14 +11,24 @@ hostname="$(kubectl get service ingress-nginx-controller -o jsonpath='{.status.l
...
+export NUODB_CP_PASSWORD="$(kubectl get secret dbaas-user-system-admin -o jsonpath='{.data.password}' | base64 -d)"
+
+# Use token authentication if server is configured to generate tokens
+if [ -n "$(kubectl get secrets nuodb-cp-runtime-config -o jsonpath='{.data.secretPassword}' --ignore-not-found)" ]; then
+    NUODB_CP_TOKEN="$(nuodb-cp httpclient POST login --jsonpath token --unquote)"
+    if [ -n "$NUODB_CP_TOKEN" ]; then
+        unset NUODB_CP_USER
+        unset NUODB_CP_PASSWORD
+    fi
+fi
```
Comment (kontaras) [inline deploy/k8s/env.sh:25]: Once we bump the minimum supported CP version to a version that supports token auth, are we going to lose automated test coverage of login with basic credentials?

Comment (adriansuarez) [reply deploy/k8s/env.sh:25]: This is one of the 3 test drivers. KWOK and External will still use basic auth.

**Review by kontaras — APPROVED** [review]

**Pull request merged**